app/Http/Controllers/Api/RequestedItemsApiController.php

```
@@ -0,0 +1,171 @@
+     * @param RequestedItems $requestedItem
+     * @return JsonResponse
+     */
+    public function update(RequestedItemsUpdateStatusRequest $request, RequestedItems $requestedItem): JsonResponse
+    {
+        $requestedItem->update([
+            'status' => $request->validated('status'),
+        ]);
+
+        return response()->json([
+            'message' => 'Requested item status updated successfully',
+        ]);
+    }
+
+    /**
+     * @OA\Delete(
+     *     path="/api/requestedItems/{requestedItem}",
+     *     summary="Delete requested item",
+     *     tags={"Requested Items"},
+     *     @OA\Parameter(
+     *         name="requestedItem",
+     *         in="path",
+     *         description="Requested item id",
+     *         required=true,
+     *         @OA\Schema(
+     *             type="integer"
+     *         )
+     *     ),
+     *     @OA\Response(response="200", description="Delete requested item", @OA\JsonContent(
+     *       ref="#/components/schemas/RequestedItems"
+     *     )
+     *   )
+     * )
+     *
+     * @param RequestedItems $requestedItem
+     * @return JsonResponse|ValidationException
+     * @throws ValidationException
+     * @throws ValidationException
+     */
+    public function delete(RequestedItems $requestedItem): JsonResponse|ValidationException
+    {
+        if ($requestedItem->status === 'on_approval') {
+            $isDeleted = $requestedItem->delete();
+
+            if (!$isDeleted) {
+                throw ValidationException::withMessages(['status' => 'Requested item was not deleted']);
+            }
+
+            RequestItemEvent::dispatch(json_decode($requestedItem->inventory_ids), false);
+
+            return response()->json([
+                'message' => 'Requested item deleted successfully',
+            ]);
+        } else {
+            throw ValidationException::withMessages([
+                'status' => 'Only items with status "on_approval" can be deleted'
```

In a php world we should return an error. Thanks that laravel magicaly transforms thows to proper response

---

better manual return 422?